Std/Data/List/Basic.lean

```diff
@@ -254,6 +254,59 @@ where
     | _::_ => by simp [intercalateTR.go, go]
     simp [intersperse, go]
 
+/-! ## Algebraic instances -/
+
+instance {α : Type _} : Std.Associative (α := List α) List.append where
+  assoc := append_assoc
+
+instance {α : Type _} : Std.LawfulIdentity (α := List α) List.append nil where
```

Comment:
I think this should be
```suggestion
instance {α : Type _} : Std.LawfulIdentity (α := List α) (· ++ ·) [] where
```
so that it works with `Append` notation.